src/components/android-settings-container.js

```diff
@@ -74,12 +85,20 @@ class AndroidSettingsContainer extends React.Component {
             return true
         }
 
+        let { numberSmsToSend, numberSmsConfirmation } = this.state
+        numberSmsConfirmation == ''
+            ? (numberSmsConfirmation = null)
+            : (numberSmsConfirmation = numberSmsConfirmation)
+        numberSmsToSend == ''
+            ? (numberSmsToSend = null)
+            : (numberSmsToSend = numberSmsToSend)
+
         const androidData = {
             metadataSync: this.state.metadataSync,
             dataSync: this.state.dataSync,
             numberSmsToSend: this.state.numberSmsToSend,
             numberSmsConfirmation: this.state.numberSmsConfirmation,
```

Comment:
You see that here you are using `this.state.numberSmsConfirmation`, so lines 92-94 wouldn't actually change the value saved.

Reply:
Yuup I see I'm not using all the re-assigning lines, I'm removing `numberSmsConfirmation` and `numberSmsToSend` and only using states